### PR TITLE
(PUP-7042) Mark strings in module_tool

### PR DESCRIPTION
**Pull request by Magisus**
This commit marks user-facing error and info strings in
`lib/puppet/module_tool.rb` and `lib/puppet/module_tool/*` for translation.

### CODE REVIEW

**Comment by Magisus** [comment]
Closing and reopening to fix CI.

**Comment by McdonaldSeanp** [comment]
👍

File: lib/puppet/module_tool/errors/shared.rb [hunk 1]
```diff
@@ -136,17 +137,17 @@ def initialize(options)
       @module_name = options[:module_name]
       @modules     = options[:installed_modules]
       @action      = options[:action]
-      super "Could not #{@action} '#{@module_name}'; module appears in multiple places in the module path"
+      super _("Could not %{action} '%{module_name}'; module appears in multiple places in the module path") % { action: @action, module_name: @module_name }
```
Comment (MosesMendoza) [inline lib/puppet/module_tool/errors/shared.rb]: could you add a translator note that `module path` here refers to a set of directories that puppet knows of to install modules in?

File: lib/puppet/module_tool/metadata.rb [hunk 2]
```diff
@@ -174,24 +174,24 @@ def validate_name(name)
 
       err = case modname
       when nil, '', :namespace_missing
-        "the field must be a namespaced module name"
+        _("the field must be a namespaced module name")
       when /[^a-z0-9_]/i
-        "the module name contains non-alphanumeric (or underscore) characters"
+        _("the module name contains non-alphanumeric (or underscore) characters")
```
Comment (MosesMendoza) [inline lib/puppet/module_tool/metadata.rb]: Wow these regexes will prevent us from having any module names with UTF-8 characters. I wonder if there's a bug for that already. cc @Iristyle

File: lib/puppet/module_tool/metadata.rb [hunk 3]
```diff
@@ -203,19 +203,19 @@ def validate_data_provider(value)
...
         unless value =~ /^[a-zA-Z][a-zA-Z0-9_]*$/
-          err = value =~ /^[a-zA-Z]/ ? 'contains non-alphanumeric characters' : 'must begin with a letter'
+          err = value =~ /^[a-zA-Z]/ ? _('contains non-alphanumeric characters') : _('must begin with a letter')
         end
       else
-        err = 'must be a string'
+        err = _('must be a string')
       end
-      raise ArgumentError, "field 'data_provider' #{err}" if err
+      raise ArgumentError, _("field 'data_provider' %{err}") % { err: err } if err
```
Comment (MosesMendoza) [inline lib/puppet/module_tool/metadata.rb]: could you add a note that `'data_provider'` is a metadata option and that this line is the beginning of the concatenation of one of the two preceding lines?

File: lib/puppet/module_tool/tar.rb [hunk 4]
```diff
@@ -11,7 +11,7 @@ def self.instance
     elsif Puppet.features.minitar? && Puppet.features.zlib?
       Mini.new
     else
-      raise RuntimeError, 'No suitable tar implementation found'
+      raise RuntimeError, _('No suitable tar implementation found')
```
Comment (MosesMendoza) [inline lib/puppet/module_tool/tar.rb]: Could you add a note that `tar` is an application dependency?

File: lib/puppet/module_tool/metadata.rb (hunk 3 above)
Comment (Magisus) [inline lib/puppet/module_tool/metadata.rb]: Gonna just refactor this whole thing. Again, pretty sure it's untranslatable as is.

**Comment by Magisus** [comment]
Updated.

File: lib/puppet/module_tool/metadata.rb (hunk 3 above)
Comment (MosesMendoza) [inline lib/puppet/module_tool/metadata.rb]: 👍

**Pull request merged**